airbyte-integrations/connectors/source-stripe/unit_tests/test_streams.py

```diff
@@ -785,6 +785,7 @@ def test_subscription_items_extra_request_params(requests_mock, stream_by_name,
                                 "created": 1699603175,
                                 "quantity": 1,
                                 "subscription": "sub_1OApco2eZvKYlo2CEDCzwLrE",
+                                'updated': 1699603884
```

Comment:
```suggestion
```

This endpoint does not return the `updated` field.
This test calls the `subscription_items` stream in `full_refresh`.
My understanding is that the `updated` field of `subscriptions` is set to the `created` value when we're in full refresh and to the update event creation time when we're in `incremental`.

This is due to the fact that the `subscriptions` is an `IncrementalStripeStream`:
According to its docstring:
*This class combines both normal incremental sync and event based sync. It uses common endpoints for sliced data syncs in the full refresh sync mode and initial incremental sync. For incremental syncs with a state, event based sync comes into action.* `subscriptions.updated` get copied to `subscriptions_items.updated` by the `slice_data_retriever`.

So for this test to pass we'd have to set the `updated` value of the subscription items records to the `created` value of the `subscription`.

What I find pretty confusing is that:
-  🟢 On `subcription_items` the `created` value is  always given by the normal api.
- 🟡 Its `updated` value corresponds to the `subscriptions` update time - and not the item update time. Which I could tolerate if we'd rename this field to `subscription_updated`.
- 🔴  When we use full refresh (or first incremental sync) we'll have `updated` always set to the `created` value of subscriptions (both on the subscription and its items).
- 🔴 When we use incremental, on subsequent syncs, we set `updated` to the timestamp of the update event.

🚨  I think this means that you get different `updated` field value according to your selected sync mode. IMO sync mode should not change the underlying synced data. I'll reach out to my team to check if this is a known and accepted behavior. If it is, I would find it more correct and less confusing to set `updated` to `null` when the sync mode is `full_refresh`.

@williamkaper let me know what you think. I might be missing something. I think your inputs as a production user of this connector would be pretty valuable.
This comment probably also applies to your other PR https://github.com/airbytehq/airbyte/pull/46864

@maxi297 I need your 🧠 to confirm I'm correct and understand if it's an accepted behavior.

Reply:
@alafanechere I would prefer to have updated set to the same value as created vs NULL in the final stream. Otherwise as a consumer, Ill need to coalesce that value to created anytime I go to use it.

The column name could be changed to subscription_updated if that is the convention in other connectors